More v1.0.0

huss opened the pull request:
# Description

This addresses the following issues:

- Make labels so they don't have group / meter in them on modal pages specific to an item. This brings consistency to the labels. Later did more to remove translations specific in many pages.
- Fix up translations so unused ones are removed.
- Fix a mistake where defaults for validation gotten for CSV upload. Fixing this caused other issues to appear because the validation was not being done correctly.
- Change curl type uploads so they use yes / no instead of true / fault to be consistent with the web pages. Note that the CSV files still use boolean values since they directly go into the database. This could be changed in the future.
- Removed many uses of true / false to be the enum that was really controlling the choice.
- Remove unused as a unit type since it was not being effectively used in the code but had been in the design. Changed uses in testing to another type since it did not matter which one.
- Added csv.gz files to git ignore because they sometimes show up from testing and are left when there is a failure.
- Remove duplicate area check for database migration.
- Fixed bug in testing code where migrations did not account for ones of major version of one and above.
- A few other small changes.

## Type of change

- [x] Note merging this changes the database configuration.
- [ ] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request
- [x] You acknowledge that every person contributing to this work has signed the [OED Contributing License Agreement](https://openenergydashboard.github.io/developer/cla.html) and each author is listed in the Description section.

## Limitations

Note that the database enum type of unit_represent_type had one option removed but current developer systems should be fine if it is left for now.

huss committed:
Change group note label so consistent with other pages

huss committed:
Remove group note translations

huss committed:
Remove duplicate area constraint from migration

huss committed:
Constraint for cumulative reset

huss committed:
Fix variable location

huss committed:
fix types + lots other

- Both web page and curl upload use yes / no instead of true / false
- remove UnitRepresentType.unused since not needed
- Uses of unused was changed to quantity in all places because value
did not matter
- Removed new cumulative check on meter because times can be
backwards when you want to start in the previous day
- Remove unused.groups and unused.meters in translations since unused
- Added the key when a CSV validation fails to message
- git ignore for .gz files added

huss committed:
Update labels

Titles for meters/groups remove those names.
Fix some formatting.

huss committed:
fix gps labels

huss committed:
Remove unneeded line

huss committed:
Bug when recalibrate a map

There was a bug that meant after recalibrating a map it did not properly
return to the map overview page. This is now fixed by explicitly going
back to the proper page. Note there is an issue that OED thinks there
was a change to the map and prompts to see if you want to save. This is
being left for now and is noted in the help pages. It may go away when
this page is converted to modals or be fixed at a future time.

huss committed:
More translation uniformity

ChrisMart21 commented on src/server/services/csvPipeline/uploadMeters.js:
```diff
@@ -27,7 +27,7 @@ async function uploadMeters(req, res, filepath, conn) {
 
 	// If there is a header row, we remove and ignore it for now.
-	const meters = (req.body.headerRow === 'true') ? temp.slice(1) : temp;
+	const meters = (req.body.headerRow === BooleanTypesJS.true) ? temp.slice(1) : temp;
```
Overall these changes look good. No issues with the formal testing, and web testing. There is a potential issue where some values used are not imported which may cause issues down the line.  Import for BoooleanTypesJS should be added.

huss replied:
@ChrisMart21 I wanted to ask about the comment on values not imported. I don't know if I know for sure what you mean but the code is supposed to use undefined if the value is not in the CSV file/parameters and then default values are used by OED. Values that cannot have default values are supposed to fail validation checks if they are not present. Does this address what you were talking about? Are there some values that seem not to follow the design?

I think I have addressed all the other comments.

ChrisMart21 replied:
You have indeed addressed my concerns, and the appropriate updates have been made.

ChrisMart21 reviewed (changes requested):
Update import, and resolve merge conflicts

huss committed:
Merge remote-tracking branch 'upstream/v1.0.0' into moreV1.0.0

huss committed:
Add missing import

huss commented:
Unfortunately the failing tests on migration seen in recent PRs are still occurring after this merge and I am looking into the issue.

ChrisMart21 commented:
> Unfortunately the failing tests on migration seen in recent PRs are still occurring after this merge and I am looking into the issue.


Interesting. I've rebuilt the container from the bottom up including node_modules, and database. I did not encounter any failing tests.

huss committed:
Fix migration tests so work for any version of OED

The migration tests did not account for a major version of 1 and above.
This fixes that and also makes them more general so they work with any
major, minor and patch version.

huss commented:
> > Unfortunately the failing tests on migration seen in recent PRs are still occurring after this merge and I am looking into the issue.
> 
> Interesting. I've rebuilt the container from the bottom up including node_modules, and database. I did not encounter any failing tests.

This only started when the version was increased to 1.0.0. That only happened after I merged in the current version of the branch v1.0.0 so @ChrisMart21 did not see it. I have updated the migration test code so it should work for all versions of OED into the future. All checks and tests pass and this is now ready to merge.
